docs/DeveloperGuide.md

```diff
@@ -364,12 +383,71 @@ This product helps users streamline the process of tracking and managing their i
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+1. Usability: The command-line interface should be intuitive and easy to use, even for non-technical users.
+2. Reliability: The application should not lose any user data during normal operation or unexpected shutdowns.
+3. Compatibility: The application should run on common operating systems (Windows, macOS, Linux).
+4. Maintainability: The code should be well-documented and follow clean code principles for easy future enhancements.
+5. Portability: User data should be easily exportable and importable for backup purposes or switching devices.
+6. Performance: The application should respond to all user commands without negatively noticable delay.
+7. Scalability: The application should handle an extensive number of internship entries (e.g., 100+ entries) without a noticeable decrease in performance.
+8. Data Persistence: All data should be automatically saved to a file, allowing users to resume their progress upon reopening the application.
+
 
 ## Glossary
 
-* *glossary item* - Definition
+* *Internship* - The position of a student or trainee who works in an organisation, with associated role, company, duration and deadline information.
+*  *InternshipList* - A collection of Internships added by the user.
+*  *Deadline* - The date which an event is schedule for.
+*  *Parser* - The component responsible for interpreting user commands and executing the appropriate actions.
+*  *Storage* - The component responsible for saving and loading the user's data to and from a file. 
 
 ## Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+1. **Installation and Setup:**
+   - Ensure Java Runtime Environment (JRE) is installed on your system.
+   - Download the EasInternship application JAR file.
+   - Open a terminal or command prompt and navigate to the directory containing the JAR file.
+
+
+2. **Running the Application:**
+   - Execute the command: java -jar EasInternship.jar
+   - Verify that the application starts and displays a welcome message.
+
+
+3. **Testing Basic Commands:**
+   - Enter the `help` command and verify that usage instructions are displayed.
+   - Test the `exit` command to ensure the application exits properly.
+
+
+4. **Adding an Internship:**
+   - Enter the command `add -role {role name} -company {company name}` eg. `add -role Software Engineer Intern -company Google`
+   - Verify that the internship has been added successfully.
+
+   
+5. **Updating an Internship:**
+  - Enter the command `update {ID} {-field} {updated information}` eg. `update 1 -skills python, java`.
```

Comment:
Under the updated information, even though it is valid to use the comma in such a way, can we not put it as an example. I think having to use commas is quite confusing for a user

Reply:
Understood! Removed.